client/src/main/kotlin/io/spine/examples/shareaware/client/Input.kt

```diff
@@ -0,0 +1,134 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.client
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.interaction.collectIsFocusedAsState
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+import io.spine.examples.shareaware.client.payment.Tooltip
+
+/**
+ * The input component that supports displaying a tip.
+ *
+ * @param value the input text to be shown in the text field
+ * @param onChange the callback that is triggered when the input's value changes
+ * @param placeholder the label to be displayed inside the input container
+ * @param isError indicates if the input's current value is in error
+ * @param tipMessage message to be displayed in the tooltip
+ * @param containerColor the color used for the background of this input
+ * @param leadingIcon the optional leading icon to be displayed at the beginning of the input field container
+ * @param contentPadding the spacing values to apply internally between the input container and the content
+ */
+@Composable
+public fun Input(
+    value: String,
+    onChange: (String) -> Unit,
+    placeholder: String,
+    isError: Boolean,
+    tipMessage: String = "",
+    containerColor: Color = MaterialTheme.colorScheme.tertiary,
+    leadingIcon: @Composable (() -> Unit)? = null,
+    contentPadding: PaddingValues = PaddingValues(
+        horizontal = 16.dp,
+        vertical = 8.dp
+    )
+) {
```

Comment:
The body of this function is too long. Please extract some meaningful bits into components (preferably) or functions.